pkg/ingester/client/compat.go

```diff
@@ -276,11 +282,57 @@ func (s Sample) MarshalJSON() ([]byte, error) {
 func (s *Sample) UnmarshalJSON(b []byte) error {
 	var t model.Time
 	var v model.SampleValue
-	vs := [...]json.Unmarshaler{&t, &v}
+	vs := [...]stdjson.Unmarshaler{&t, &v}
 	if err := json.Unmarshal(b, &vs); err != nil {
 		return err
 	}
 	s.TimestampMs = int64(t)
 	s.Value = float64(v)
 	return nil
 }
+
+func init() {
+
+	jsoniter.RegisterTypeEncoderFunc("client.Sample", func(ptr unsafe.Pointer, stream *jsoniter.Stream) {
+		sample := (*Sample)(ptr)
+
+		stream.WriteArrayStart()
+		stream.WriteRaw(model.Time(sample.TimestampMs).String())
```

Comment:
Why not `stream.WriteFloat64()` ?

Reply:
We could (and divide by milliseconds, which is done by model.Time.String())